v2/protocol/http/protocol.go

```diff
@@ -133,6 +145,64 @@ func (p *Protocol) Request(ctx context.Context, m binding.Message) (binding.Mess
 	return NewMessage(resp.Header, resp.Body), NewResult(resp.StatusCode, "%w", result)
 }
 
+func (p *Protocol) doWithRetry(backoff *types.Backoff) func(*http.Request) (binding.Message, error) {
```

Comment:
I think this needs context.Context passed in to allow for the app to cancel in the case of shutdown or term signals.

Reply:
good point! thanks!